src/lib.rs

```diff
@@ -654,29 +658,68 @@ impl Maildir {
         // to new dir. see for example http://www.courier-mta.org/maildir.html.
         let pid = std::process::id();
         let hostname = gethostname::gethostname();
+        let hostname = hostname.to_string_lossy();
```

Comment:
Combine these two into one line `gethostname::gethostname().to_string_lossy()`

Reply:
I wanted to do that, but:

```
error[E0716]: temporary value dropped while borrowed
   --> src/lib.rs:660:24
    |
660 |         let hostname = gethostname::gethostname().to_string_lossy();
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^                  - temporary value is freed at the end of this statement
    |                        |
    |                        creates a temporary value which is freed while still in use
...
679 |             tmppath.push(format!("{secs}.X{counter:x}M{nanos}P{pid}.{hostname}"));
    |                                                                      -------- borrow later used here
    |
help: consider using a `let` binding to create a longer lived value
    |
660 ~         let binding = gethostname::gethostname();
661 ~         let hostname = binding.to_string_lossy();
```

I'll use an alternative approach that avoids the shadowed local variable.